### PR TITLE
Add support for imjournal rate limit settings 

### PR DESCRIPTION
**Pull request by shaunrampersad**
Fixes issue #200

### CODE REVIEW

**Comment by madAndroid** [comment]
Would be great if this could be merged in, we're looking for this particular config as well.

File: templates/rsyslog.conf.erb [hunk 1]
```diff
@@ -67,6 +67,13 @@ $Umask <%= scope.lookupvar('rsyslog::umask') %>
 $OmitLocalLogging on
 <% end -%>
 
+# Settings for imjournal
+<% if scope.lookupvar('rsyslog::im_journal_statfile') -%>
+$IMJournalStateFile <%=scope.lookupvar('rsyslog::im_journal_statfile') %>
```
Comment (madAndroid) [inline templates/rsyslog.conf.erb]: This doesn't appear to be a valid parameter - perhaps it needs to be added to `manifests/init.pp` and `manifests/params.pp` ?

Comment (shaunrampersad) [inline templates/rsyslog.conf.erb]: Correct. Not sure how I missed this. Will fix

Comment (madAndroid) [inline templates/rsyslog.conf.erb]: Cool, thanks :) I've got these changes in a test env at the moment, all looking good - it's the only way I could get a newer format JSON template to work.

Comment (shaunrampersad) [inline templates/rsyslog.conf.erb]: I've updated. Hopefully this fixes it. :)

File: templates/rsyslog.conf.erb [hunk 2]
```diff
@@ -67,6 +67,14 @@ $Umask <%= scope.lookupvar('rsyslog::umask') %>
 $OmitLocalLogging on
 <% end -%>
 
+# Settings for imjournal
+<% if scope.lookupvar('rsyslog::im_journal_statefile') -%>
+$IMJournalStateFile <%=scope.lookupvar('rsyslog::im_journal_statefile') %>
+<% end %>
+$ImjournalIgnorePreviousMessages <%=scope.lookupvar('rsyslog::im_journal_ignore_previous_messages') %>
```
Comment (madAndroid) [inline templates/rsyslog.conf.erb]: This field, and the new ones added below, will come up blank on operating systems without journald (which rsyslog might not handle properly), unless they're wrapped with a `scope.lookupvar` (much like the other parameters).

Comment (shaunrampersad) [inline templates/rsyslog.conf.erb]: yes you're right. Fixed.

Comment (madAndroid) [inline templates/rsyslog.conf.erb]: Awesome :) I'm not a maintainer at all, so won't be able to merge the PR - perhaps when @saz or another maintainer has a chance, this PR could be merged in.

**Comment by wernermaes** [comment]
Will this be merged in sometime?

**Comment by wernermaes** [comment]
This does not build?

**Pull request merged**